Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,230 @@
+
+## Future Directions
+
+### Typed captures
+
+Typed captures would statically represent how many captures and of what kind are present in a regex literals. They could produce a `Substring` for a regular capture, `Substring?` for a zero-or-one capture, and  `Array<Substring>` (or a lazy collection) for a zero(or one)-or-more capture. These are worth exploring, especially in the context of the [start of variadic generics][variadics] support, but are outside the scope of this pitch.
+
+### Raw regex literals
+
+Similar to raw string literals `#"I said "hello there""#`, we may allow additional `#` characters to be added to the regex literal delimiters to allow for easy escaping of the `/` character within the literal, e.g:
+
+```swift
+#/foo/bar/baz/#
+```
+
+This is outside of the scope of this pitch, but would make sense as an additive proposal to regex literals.
+
+### Further extension to Swift language constructs (better name?)
+
+TODO: I'm referring to things like a Regex-backed `enum` for declaring tokens.
+
+TODO: Mention restructuring pattern-matching enhancements
+
+
+### Semantic details
+
+Further details about the semantics of regex literals, such as what definition we give to character classes, the initial supported feature set, and how to switch between grapheme-semantic and scalar-semantic usage, is outside the scope of this pitch and worthy of their own discussions. 
+
+## Alternatives considered
+
+### Using a different delimiter to `/`
+
+As explored above, using `/` as the delimiter has the potential to conflict with existing operators using that character, and may necessitate:
+
+- Changing of parsing rules around chained `/` over multiple lines
+- Deprecating custom operators containing the `/` character (not including the division operator) 
+- Requiring a new language version mode to parse the literal with `/` delimiters
+
+However one of the main goals of this pitch is to introduce a familiar syntax for regular expression literals, which has been the motivation behind e.g using the PCRE regex syntax. Given the fact that `/` is an existing term of art for regular expressions, we feel that if the aforementioned parsing issues can be solved in a satisfactory manner, we should prefer it as the delimiter.
+
+### Raku-style syntax
+
+TODO: rejected as we want familiarity. If we're going to do something custom or different, then we'd like to design something beyond regex. At that point, we're talking about something pretty different (and certainly not tied to strings, e.g. data processing literals), and this word does not preclude improvements there.
+
+### Fully custom literals
+
+It may be worth exploring the ability for users to define custom literals, where they have full generality and no default type. This would require fleshing out the library-driven compilation story (bootstrapping as well as compiler and type system API). However that it is mostly orthogonal to this effort.
```

Comment:
```suggestion
It may be worth exploring the ability for users to define custom literals, where they have full generality and no default type. This would require fleshing out the library-driven compilation story (bootstrapping as well as compiler and type system API). However that it is mostly orthogonal to this effort.
```

And I think I want to address partially-custom delimiters in some fashion as well.